### PR TITLE
CORE-663 MCQ branching configuration

### PR DESCRIPTION
**Pull request by object-required**

### CODE REVIEW

File: cypress/e2e/undo-redo/block-configs-should-update-flow-state/mcq-block-configs.cy.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    beforeEach(() => {
+      cy.createFlow({
+        label: 'mcq block test',
+      })
+
+      cy.addBlock(['Select One Response'])
+
+      cy.get('[data-cy="choices-builder"]')
+        .find('[data-cy="resource--text"]')
+        .last()
```
Comment (safydy-r) [inline cypress/e2e/undo-redo/block-configs-should-update-flow-state/mcq-block-configs.cy.js:15]: Can we avoid `last()` ? this is a bit fragile, especially if we reorganize the UI. I'd suggest we add a dynamic `data-cy` on choices like ```data-cy="`choices-${uuid}`"``` or something. Your thoughts

File: src/components/interaction-designer/block-editors/choices/ChoiceMappingModal.vue [hunk 2]
```diff
@@ -1,7 +1,11 @@
 <template>
   <div class="choice-mapping-modal">
     <div class="form-group">
-      <button :disabled="block.config.choices.length === 0" class="btn btn-primary btn-sm" @click="showModal">
+      <button
+        :disabled="block.config.choices.length === 0"
+        class="btn btn-primary btn-sm"
+        data-cy="set-choice-options"
```
Comment (safydy-r) [inline src/components/interaction-designer/block-editors/choices/ChoiceMappingModal.vue:7]: add postfix `--btn` according to our convention https://app.getguru.com/folders/iAnRr8nT/Automated-testing?activeCard=59e5bfdf-a2fb-48bf-a805-cd8083b929aa

File: cypress/e2e/undo-redo/block-configs-should-update-flow-state/mcq-block-configs.cy.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    const getStore = () => cy.window().its('store')
+    const flowModuleState = () => getStore().its('state.flow')
+    const flowsListState = () => flowModuleState().its('flows')
+
+    const firstBlockState = () => flowsListState().its('[0].blocks[0]')
+
+    firstBlockState().its('config.minimum_choices').should('eq', MINIMUM_CHOICES)
+    firstBlockState().its('config.maximum_choices').should('eq', MAXIMUM_CHOICES)
+  })
+})
```
Comment (bulatgab) [inline cypress/e2e/undo-redo/block-configs-should-update-flow-state/mcq-block-configs.cy.js:176]: It was pretty complex 🙂

LGTM

**Review by bulatgab — APPROVED** [review]

File: cypress/e2e/undo-redo/block-configs-should-update-flow-state/mcq-block-configs.cy.js (hunk 1 above)
Comment (object-required) [reply cypress/e2e/undo-redo/block-configs-should-update-flow-state/mcq-block-configs.cy.js]: It's literally the last element, though I added `data-cy="choices-builder--editor-new"` to make it as explicit as possible.

Comment (safydy-r) [reply cypress/e2e/undo-redo/block-configs-should-update-flow-state/mcq-block-configs.cy.js]: I see what you mean, sorry about the confusion. Thanks for making it more explicit

**Review by safydy-r — APPROVED** [review]
It is much better now, thanks for the changes

**Pull request merged**